singleFrameProcCoreTasks/charimg.rst

```diff
@@ -0,0 +1,304 @@
+
+See `lsstDebug.info`_ for more on the debugging framework.
+
+.. _`lsstDebug.info`: https://lsst-web.ncsa.illinois.edu/doxygen/x_masterDoxyDoc/classlsst_debug_1_1_info.html 
+
+
+
+Examples
+========
+
+.. This example is not working in the current stack (see https://jira.lsstcorp.org/browse/DM-9142)  --- 2/9/2017
+   
+
+Note: running this example currently requires that over and above the DM Stack installation, `afwdata`_ is installed and set up (via the EUPS `setup <https://dev.lsstcorp.org/trac/wiki/EupsTutorial>`_ command).
+.. This is a general link to the EUPS tutorial, but setup is explained in there
+.. _`afwdata`: https://github.com/lsst/afwdata
+
+This example script is `calibrateTask.py` (which calls :doc:`CharacterizeImageTask <apiUsage_charimg>`) before calling :doc:`CalibrateTask <calibimg>` in the `$PIPE_TASKS/examples` directory, and can be run from the command line as, e.g.:
+
+
+.. code-block:: python
+  
+     python examples/calibrateTask.py -display
+
+Where the `-display` flag tells the script to bring up the display tool to show the image files after each step.
+     
+The first thing the example does is import the task (there are some other standard imports as well that are not extracted out here):
+
+.. code-block:: python
+		
+    from lsst.pipe.tasks.characterizeImage import CharacterizeImageTask
+
+The script next processes the data. This occurs in two steps:
+
+- Characterize the image: measure bright sources, fit a background and PSF, and repairs cosmic rays
+
+.. code-block:: python
+		
+     exposure = loadData()
+     exposureIdInfo = ExposureIdInfo(expId=1, expBits=5)
+ 
+     # characterize the exposure to repair cosmic rays and fit a PSF model
+     # display now because CalibrateTask modifies the exposure in place
+     charRes = charImageTask.characterize(exposure=exposure, exposureIdInfo=exposureIdInfo)
+     if display:
+         displayFunc(charRes.exposure, charRes.sourceCat, frame=1)
+
+- Calibrate the exposure: measure faint sources, fit an improved WCS and photometric zero-point
+		
+.. code-block:: python
+
+   
+    # calibrate the exposure
+    calRes = calibrateTask.calibrate(exposure=charRes.exposure, exposureIdInfo=exposureIdInfo)
+    if display:
+        displayFunc(calRes.exposure, calRes.sourceCat, frame=2)
+
+To round out this minimal description, the `displayFunc` that is called above in the blocks is defined as so:
+
+.. code-block:: python
+		
+ def displayFunc(exposure, sourceCat, frame):
+    display = afwDisplay.getDisplay(frame)
+    display.mtv(exposure)
+
+    with display.Buffering():
+        for s in sourceCat:
+            xy = s.getCentroid()
+            display.dot('+', *xy, ctype=afwDisplay.CYAN if s.get("flags_negative") else afwDisplay.GREEN)
+	
+     
+
+Algorithm details
+=================
+
```

Comment:
Hm, little unclear on this -- though i link to code if they want to go look up specifics, my description of all the techniques is pretty general i thought.  

But yes, clearly I haven't inserted any math yet anywhere (and not certain where I would do this anywhere except for say the B-F correction where we do have details..)